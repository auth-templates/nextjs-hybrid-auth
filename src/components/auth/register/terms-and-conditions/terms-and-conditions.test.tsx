import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import TermsAndConditions from './terms-and-conditions';

describe('TermsAndConditions', () => {
    test('if all elements are present', () => {
        render(<TermsAndConditions />)
        const checkbox = screen.getByRole('checkbox');
        expect(checkbox).toBeInTheDocument();
        expect(checkbox).toHaveAttribute('required');
        expect(screen.getByText(/I agree with the /)).toBeInTheDocument();
        const link = screen.getByRole('link', {name: "terms and conditions"});
        expect(link).toBeInTheDocument();
        expect(link).toHaveAttribute('href', '/terms-and-conditions');
    }),

    it('renders correctly', () => {
        const tree = renderer.create(<TermsAndConditions />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})