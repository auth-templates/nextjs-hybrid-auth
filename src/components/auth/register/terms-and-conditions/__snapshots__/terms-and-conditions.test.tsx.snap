// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TermsAndConditions renders correctly 1`] = `
<div
  className="termsAndConditions"
>
  <div
    className="container"
  >
    <input
      checked={false}
      id="checkbox"
      name="checkbox"
      onChange={[Function]}
      required={true}
      type="checkbox"
    />
    <span
      className="checkmark"
      data-testid="checkbox-cover"
      onClick={[Function]}
    />
  </div>
  <div>
    I agree with the 
    <a
      href="/terms-and-conditions"
    >
      terms and conditions
    </a>
    .
  </div>
</div>
`;
